packages/blade/src/components/Tooltip/Tooltip.web.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+import {
+  shift,
+  FloatingPortal,
+  arrow,
+  flip,
+  FloatingArrow,
+  offset,
+  useDelayGroup,
+  useDelayGroupContext,
+  useFloating,
+  useFocus,
+  useHover,
+  useInteractions,
+  useRole,
+  useTransitionStyles,
+  autoUpdate,
+} from '@floating-ui/react';
+import React from 'react';
+import { TooltipProps } from './types';
+import { TooltipContent } from './TooltipContent';
+import { ARROW_HEIGHT, ARROW_WIDTH } from './constants';
+import { getPlacementParts } from './utils';
+import { useTheme } from '~components/BladeProvider';
+import BaseBox from '~components/Box/BaseBox';
+import { makeAccessible } from '~utils';
+import { useId } from '~src/hooks/useId';
+import { size } from '~tokens/global';
+
+const Tooltip = ({
+  content,
+  children,
+  placement = 'top',
+  onOpenChange,
+}: TooltipProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const id = useId();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const arrowRef = React.useRef<SVGSVGElement>(null);
+
+  const GAP = theme.spacing[2];
+  const [side] = getPlacementParts(placement);
+  const isHorizontal = side === 'left' || side === 'right';
+  const isOppositeAxis = side === 'right' || side === 'bottom';
+
+  const { refs, floatingStyles, context } = useFloating({
+    placement,
+    open: isOpen,
+    strategy: 'fixed',
+    onOpenChange: (open) => {
+      if (open) {
+        setIsOpen(true);
+        onOpenChange?.({ isOpen: open });
+      } else {
+        setIsOpen(false);
+        onOpenChange?.({ isOpen: open });
+      }
+    },
+    middleware: [
+      shift({ crossAxis: false, padding: GAP }),
```

Comment:
what is this padding for?

Reply:
So that the tooltip never touches the edge of the screen, it will always have a 4px padding.